ui/pages/smart-transactions/smart-transaction-status-page/smart-transaction-status-page.tsx

```diff
@@ -0,0 +1,434 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  SmartTransactionStatuses,
+  SmartTransaction,
+} from '@metamask/smart-transactions-controller/dist/types';
+
+import {
+  Box,
+  Text,
+  Icon,
+  IconName,
+  IconSize,
+  Button,
+  ButtonVariant,
+  ButtonSecondary,
+} from '../../../components/component-library';
+import {
+  AlignItems,
+  BlockSize,
+  BorderStyle,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextVariant,
+  TextColor,
+  FontWeight,
+  IconColor,
+  TextAlign,
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { getCurrentChainId } from '../../../selectors';
+import { getFeatureFlagsByChainId } from '../../../../shared/modules/selectors';
+import { BaseUrl } from '../../../../shared/constants/urls';
+import { hideLoadingIndication } from '../../../store/actions';
+import { hexToDecimal } from '../../../../shared/modules/conversion.utils';
+
+type RequestState = {
+  smartTransaction?: SmartTransaction;
+  isDapp: boolean;
+};
+
+export type SmartTransactionStatusPageProps = {
+  requestState: RequestState;
+  onCloseExtension: () => void;
+  onViewActivity: () => void;
+};
+
+const SMART_TRANSACTIONS_EXPECTED_DEADLINE_FALLBACK = 45;
+const SMART_TRANSACTIONS_MAX_DEADLINE_FALLBACK = 150;
+
+export const showRemainingTimeInMinAndSec = (
+  remainingTimeInSec: number,
+): string => {
+  if (!Number.isInteger(remainingTimeInSec)) {
+    return '0:00';
+  }
+  const minutes = Math.floor(remainingTimeInSec / 60);
+  const seconds = remainingTimeInSec % 60;
+  return `${minutes}:${seconds.toString().padStart(2, '0')}`;
+};
+
+const getDisplayValues = ({
```

Comment:
Minor, I'm a fan of using the `function` keyword for stand alone functions outside of a component.

Reply:
Why not use the arrow functions as much as possible, since it has a few benefits, including shorter syntax?